pkgs/development/interpreters/acl2/default.nix

```diff
@@ -1,50 +1,46 @@
-{ lib, stdenv, callPackage, fetchFromGitHub, writeShellScriptBin, substituteAll
+{ lib, stdenv, callPackage, fetchFromGitHub, runCommandLocal, makeWrapper, substituteAll
 , sbcl, bash, which, perl, nettools
-, openssl, glucose, minisat, abc-verifier, z3, python2
+, openssl, glucose, minisat, abc-verifier, stp, z3, python
+, libipasir ? callPackage ./libipasirglucose4 { }
```

Comment:
Please add the package to the global scope.

Reply:
I'm not sure if this is worth creating a global scoped package for - it's a one-time source code dump that doesn't have any kind of release process, repository, web page, or anything. Nothing in nixpkgs uses it except ACL2, and I guess nothing probably will in the future either... Even ACL2 might replace it with some other more up-to-date IPASIR implementation (I just chose this one because it's mentioned in the docs as an example, though it's 4 years old). Are you sure I should add this library to global scope?

Reply:
Then please don't put it intothe inputs and move it into a let in.

Reply:
Not sure it is better if someone actually finds a compatible better implementation and wants to try it…

Reply:
Before this PR it was already present as an attribute of the derivation (i.e. you could access it via `acl2.libipasirglucose4`), so it was already possible to override it with `acl2.overrideAttrs`. I just wanted to make it easier by putting it as an input so it could be swapped out with `acl2.override`, but I'm fine with putting it back as an attribute like before.

Reply:
I changed it so it's an attribute again. I also added a comment explaining how to override it. Now you can also override it at runtime using an environment variable, which is the upstream-sanctioned way of doing it anyway since the library is loaded with dlopen.